database/db.go

```diff
@@ -0,0 +1,79 @@
+package database
+
+import (
+	"database/sql"
+	"errors"
+	"fmt"
+	"sync"
+
+	_ "github.com/lib/pq"
+
+	"durn2.0/conf"
+	"durn2.0/durn"
+)
+
+var mutex sync.Mutex
+var db *sql.DB
+
+// CreateDBConnection Initializes the database connection using
+// the connection details specified in env-vars
+func CreateDBConnection() {
+	c := conf.ReadConfiguration()
+	psqlconn := fmt.Sprintf("host=localhost port=%d user='%s' password='%s' dbname='%s' sslmode=disable", c.DBPort, c.DBUser, c.DBPassword, c.DBName)
+
+	mutex.Lock()
+	defer mutex.Unlock()
+	var err error
+	db, err = sql.Open("postgres", psqlconn)
+	if err != nil {
+		panic(err)
+	}
+}
+
+func DisconnectDB() {
+	mutex.Lock()
+	defer mutex.Unlock()
+
+	db.Close()
+}
+
+func QueryAllVoters() ([]durn.Voter, error) {
+	mutex.Lock()
+	defer mutex.Unlock()
+
+	query := `SELECT * from valid_voters`
+	rows, err := db.Query(query)
+	if err != nil {
+		return nil, err
+	}
+	defer rows.Close()
+	var res []durn.Voter
+
+	for rows.Next() {
+		var username durn.Voter
+		err = rows.Scan(&username)
+		res = append(res, username)
+	}
+
+	return res, nil
+}
+
+func InsertElection(e durn.Election) error {
+	query := `INSERT INTO Elections(id, name, published, finalized, opentime, closetime) values ($1, $2, $3, $4, $5, $6)`
+	_, err := db.Exec(query, e.Id, e.Name, e.IsOpen, e.IsFinalized, e.OpenTime, e.CloseTime)
```

Comment:
Just checking, this is a prepared statement?

Reply:
From what I could see, this was the case, but I will doublecheck

Reply:
This does indeed seems to be the case. While there are a separate Prepare statement that you can use manually. Both `Query` and `Exec` seems to use prepared state both according to this [blog](https://aloksinhanov.medium.com/query-vs-exec-vs-prepare-in-golang-e7c49212c36c), and this more official [documentation](https://pkg.go.dev/database/sql/driver#Execer) and [site](http://go-database-sql.org/prepared.html)